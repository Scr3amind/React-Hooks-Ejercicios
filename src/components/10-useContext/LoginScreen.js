import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button 
                className="btn btn-dark" 
                onClick={() => setUser(
                    {
                        id:'123',
                        user:'Isaac', 
                    }
                )}
            >
                Login
            </button>
        </div>
    )
}
