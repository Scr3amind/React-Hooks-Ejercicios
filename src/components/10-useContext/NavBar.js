import React from 'react';
import {Link, NavLink} from 'react-router-dom';

export const NavBar = () => {
    return (
        // <nav>
        //     <ul>
        //         <li>
        //             <Link to="./"> Home </Link>
        //         </li>
        //         <li>
        //             <Link to="./about"> About </Link> 
        //         </li>
        //         <li>
        //             <Link to="./login"> Login </Link> 
        //         </li>
        //     </ul>
        // </nav>
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link className="navbar-brand" to="./" >useContext</Link>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" className="nav-link" to="./">Home</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link" to="./about">About</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link" to="./login">Login</NavLink>
                </div>
            </div>
        </nav>
    )
}
