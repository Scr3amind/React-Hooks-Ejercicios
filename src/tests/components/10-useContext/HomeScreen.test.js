import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/10-useContext/HomeScreen";
import { UserContext } from "../../../components/10-useContext/UserContext";

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name: 'Isaac',
        email: 'mail@mail.com'
    }

    const wrapper = mount( //mount renderiza todo a diferencia de shallow
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
