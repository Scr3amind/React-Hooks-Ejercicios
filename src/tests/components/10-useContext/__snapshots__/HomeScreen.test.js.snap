// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Isaac",
   "email": "mail@mail.com"
}
  </pre>
</div>
`;
